Assets/Plugins/Support/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,123 @@
+using Support;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using NaughtyAttributes;
+namespace Support { 
+    public class AudioManager : MonoSingleton<AudioManager>
+    {
+        //getting resources
+        [Foldout("Source")]
+        [SerializeField]
+        [ReorderableList]
+        private Sound[] sounds;
+
+        [Foldout("Source")]
+        [SerializeField]
+        [ReorderableList]
+
+        private Sound[] music;
+        [Foldout("Source")]
+        [SerializeField]
+        [ReorderableList]
+        private Sound[] dialogs;
+        //Dictionary - fetch resources from Arrays to simplisify choosing a sound
+        private Dictionary<string,Sound> _sounds = new Dictionary<string, Sound>();
+        private Dictionary<string,Sound> _music = new Dictionary<string, Sound>();
+        private Dictionary<string,Sound> _dialogs = new Dictionary<string, Sound>();
+
+        private void Awake()
+        {
+            CopySoundsListIntoDictionary(dialogs, _dialogs);
+            CopySoundsListIntoDictionary(music,_music );
+            CopySoundsListIntoDictionary(sounds,_sounds );
+        }
+
+        #region PRIVATE_METHODS
+        private void CopySoundsListIntoDictionary(Sound[] arr, Dictionary<string, Sound> dic)
+        {
+            foreach(Sound s in arr)
+            {
+                s.Source = gameObject.AddComponent<AudioSource>();
```

Comment:
You should not assign components externally - it is not safe. Check AudioSource comments for more.